lib/hawkular/metrics/metric_api.rb

```diff
@@ -191,19 +203,15 @@ def raw_data(ids, starts: nil, ends: nil, limit: nil, order: nil)
       # @param bucketDuration [String] optional interval (default no aggregation)
       # @return [Array[Hash]] datapoints
       # @see #push_data #push_data for datapoint detail
-      def get_data_by_tags(tags, starts: nil, ends: nil, bucketDuration: nil)
-        params = { tags: tags_param(tags), start: starts,
-                   end: ends, bucketDuration: bucketDuration }
+      def get_data_by_tags(tags, starts: nil, ends: nil, bucketDuration: nil, buckets:nil)
+        params = { tags: @client.tags_param(tags), start: starts,
+                   end: ends, bucketDuration: bucketDuration, buckets: buckets }
         path = "/#{@resource}/"
         @legacy_api ? path << 'data/?' : path << 'stats/?'
         resp = @client.http_get(path + encode_params(params))
         resp.is_a?(Array) ? resp : [] # API returns no content (empty Hash) instead of empty array
       end
 
-      def tags_param(tags)
```

Comment:
I need to use it at the level of the Client class. so I moved it to that class in order to reuse the code.